neuraxle/steps/flow.py

```diff
@@ -0,0 +1,133 @@
+        """
+        Send expanded data container to the wrapped handle_transform method, and returned the reduced transformed data container (back to it's orginal shape).
+
+        :param data_container: data container to transform
+        :type data_container: DataContainer
+        :param context: execution context
+        :type context: ExecutionContext
+        :return: data container
+        :rtype: DataContainer
+        """
+        expanded_data_container = self._create_expanded_data_container(data_container)
+
+        expanded_data_container =  self.wrapped.handle_transform(
+            expanded_data_container,
+            context.push(self.wrapped)
+        )
+
+        return expanded_data_container.reduce_dim()
+
+    def handle_fit_transform(self, data_container: DataContainer, context: ExecutionContext):
+        """
+        Send expanded data container to the wrapped handle_fit_transform method,
+        and returned the reduced transformed data container (back to it's orginal shape).
+
+        :param data_container: data container to fit_transform
+        :type data_container: DataContainer
+        :param context: execution context
+        :type context: ExecutionContext
+        :return: data container
+        :rtype: DataContainer
+        """
+        expanded_data_container = self._create_expanded_data_container(data_container)
+
+        self.wrapped, expanded_data_container = self.wrapped.handle_fit_transform(
+            expanded_data_container,
+            context.push(self.wrapped)
+        )
+
+        return self, expanded_data_container.reduce_dim()
+
+    def handle_fit(self, data_container: DataContainer, context: ExecutionContext):
+        """
+        Send expanded data container to the wrapped handle_fit method,
+        and returned the reduced transformed data container (back to it's orginal shape).
+
+        :param data_container: data container to fit_transform
+        :type data_container: DataContainer
+        :param context: execution context
+        :type context: ExecutionContext
+        :return: data container
+        :rtype: DataContainer
+        """
+        expanded_data_container = self._create_expanded_data_container(data_container)
+
+        self.wrapped, expanded_data_container = self.wrapped.handle_fit(
+            expanded_data_container,
+            context.push(self.wrapped)
+        )
+
+        return self, expanded_data_container.reduce_dim()
+
+    def _create_expanded_data_container(self, data_container: DataContainer) -> ExpandedDataContainer:
+        """
+        Create expanded data container.
+
+        :param data_container: data container to expand
+        :type data_container: DataContainer
+        :return: expanded data container
+        :rtype: ExpandedDataContainer
+        """
+        current_ids = self.hash(data_container.current_ids, self.hyperparams, data_container.data_inputs)
```

Comment:
Hey it's very weird that BaseStep needs its own hyperparameters passed to `.hash`. Too many arguments. We should only pass the data container. 

Important: It's a different hash method than `BaseHasher.hash`. It should have been named differently earlier. It's like `save` and `save_step`. Here maybe `hash` and `hash_chain` ?? idk. 

Please open issue if not fixed here.

Reply:
I agree, I have almost made that change, but I wanted to keep my changes small in this pr

Reply:
done anyways aa2aa56